docs/documentation/siddhi-4.0.md

```diff
@@ -1754,6 +1754,176 @@ define stream TempStream (deviceID long, roomNo int, temp double);
+attribute that matches the value for the `roomNumber` attribute of the selection which has 10 as the actual value.
+
+```sql
+select 10 as roomNumber
+delete RoomTypeTable
+on RoomTypeTable.roomNo == roomNumber;
+```
+
+### Update
+
+Update store query can be used to update selected attributes stored in a table based on a condition.
+
+**Syntax**
+
+```sql
+select <attribute name>, <attribute name>, ...?
+update <table>
+    set <table>.<attribute name> = (<attribute name>|<expression>)?, <table>.<attribute name> = (<attribute name>|<expression>)?, ...
+    on <condition>
+```
+
+The `condition` element specifies the basis on which records are selected to be updated.
+When specifying the `condition`, table attributes must be referred to with the table name.
+
+You can use the `set` keyword to update selected attributes from the table. Here, for each assignment, the attribute specified in the left must be the table attribute, and the one specified in the right can be a stream/table attribute a mathematical operation, or other. When the `set` clause is not provided, all the attributes in the table are updated.
+
+
+!!! note
+    Table attributes must be always referred to with the table name as, 
+     `<table name>.<attibute name>`.
+
+**Example**
+
+The following query updates the room occupancy by increasing its value by 1, in the `RoomOccupancyTable` table for each room number greater than 10.
+
+```sql
+select 10 as roomNumber, 1 as arrival
+update RoomTypeTable
+    set RoomTypeTable.people = RoomTypeTable.people + arrival
+    on RoomTypeTable.roomNo == roomNumber;
+```
+
+### Update or Insert Into
+
+This allows you to update the given attributes, if a matching record to the given condition already exists in the database table.  
+Or else insert the entry as a new record.
+
+**Syntax**
+
+```sql
+select <attribute name>, <attribute name>, ...
+update or insert into <table>
+    set <table>.<attribute name> = <expression>, <table>.<attribute name> = <expression>, ...
+    on <condition>
+```
+The `condition` element specifies the basis on which records are selected for update.
+When specifying the `condition`, table attributes should be referred to with the table name.
+If a record that matches the condition does not already exist in the table, the arriving event is inserted into the table.
+
+The `set` clause is only used when an update is performed during the insert/update operation.
+When `set` clause is used, the attribute to the left is always a table attribute, and the attribute to the right can be a stream/table attribute, mathematical
+operation or other. The attribute to the left (i.e., the attribute in the event table) is updated with the value of the attribute to the right if the given condition is met. When the `set` clause is not provided, all the attributes in the table are updated.
+
+!!! note
+    Table attributes should be always referred to with the table name as `<table name>.<attibute name>`.
```

Comment:
**Suggestion**: "Table attributes must always be referred to with the table name as shown below:
`<table_name>.<attribute_name>`